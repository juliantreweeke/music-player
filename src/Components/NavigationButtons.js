import React from "react";
import styled from "styled-components";

const Layout = styled.div`
  display:flex;
  width:100%;
  margin-left:20%;
  margin-right:20%;
  margin-top:auto;
  height: 25%;
  background: transparent;
  align-items:center;
  justify-content: space-between;
  &svg {
      border:2px solid red;
  }
`;

const NextButton = () => (
    <svg version="1.1" width="32" height="32" viewBox="0 0 16 16">
        <path fill="#70C1B3" d="M2 1v14l10-7z"></path>
        <path fill="#70C1B3" d="M12 1h2v14h-2v-14z"></path>
    </svg>
);

export const NavigationButtons = () => {
    return (
        <Layout>
            <NextButton/>
            <NextButton/>
        </Layout>)
}