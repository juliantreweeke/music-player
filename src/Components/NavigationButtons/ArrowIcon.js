import React from "react";
import styled from 'styled-components/macro'

const svgStyles = (data) => 
    `z-index:10;
    ${data && `cursor:pointer`};
    path {
        fill:${data ? '#79D9C9' : '#333738'};
    };
    &:hover path {
        fill:${data ? 'white' : '#333738'};
    };
    path {
        transition:fill 0.5s ease;
    }
    
`

export const ArrowIcon = ({ data, handleClick, rotate180}) => {
    return (
        <svg 
            css={svgStyles(data)} 
            onClick={handleClick} 
            transform={rotate180 && 'rotate(180)'} 
            width="32" height="32" 
            viewBox="0 0 16 16"
        >   
            <path d="M2 1v14l10-7z"></path>
            <path d="M12 1h2v14h-2v-14z"></path>
        </svg>
    );
} 
