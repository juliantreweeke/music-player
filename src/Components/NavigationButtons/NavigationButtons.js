import React from "react";
import styled from 'styled-components/macro'
import { ArrowIcon } from './ArrowIcon';
import { useTracks } from '../../Hooks/useTracks.js';
import { usePlaying } from '../../Hooks/usePlaying.js';


const Layout = styled.div`
  display:flex;
  width:60%;
  margin-left:auto;
  margin-right:auto;
  margin-top:auto;
  height:200px;
  background: transparent;
  align-items:center;
  justify-content: space-between;
`;

const BackButton = ({data, selectedTrack}) => {
    const { setSelectedTrack } = useTracks();
    const { setPlay } = usePlaying();

    const previousTrack = () => {
        if(selectedTrack === 0){
            setSelectedTrack(data.length - 1)
          } else {
            setSelectedTrack(selectedTrack -= 1 )
          }
          setPlay();
    }
    return (
        <ArrowIcon 
        data={data}
        handleClick={data && previousTrack}
        rotate180={true}
        />    
    ) 
} 

const NextButton = ({data, selectedTrack}) => {
    const { setSelectedTrack } = useTracks();
    const { setPlay } = usePlaying();

    const nextTrack = () => {
        if(selectedTrack === data.length - 1){
          setSelectedTrack(0);
        } else {
          setSelectedTrack(selectedTrack += 1);
        }
        setPlay();
    }
    return (
        <ArrowIcon 
        data={data}
        handleClick={data && nextTrack}
        />
    ) 
} 

export const NavigationButtons = ({data,selectedTrack}) => {
    return (
        <Layout>
            <BackButton data={data} selectedTrack={selectedTrack}/>
            <NextButton data={data} selectedTrack={selectedTrack}/>
        </Layout>)
}