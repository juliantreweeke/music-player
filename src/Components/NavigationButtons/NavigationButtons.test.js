import React from 'react';
import { NavigationButtons } from './NavigationButtons';
import { cleanup, render} from '@testing-library/react'
import { store } from '../../redux/store/index.js';
import { Provider } from 'react-redux';

const mockData = [{ title: 'song1', index: 0 }, { title: 'song2', index: 1 }]

describe('NavigationButtons', () => {
    afterEach(cleanup)     
    
    it("should render", () => {   
         const NavigationButtonComponent = render(
                <Provider store={store}>
                    <NavigationButtons data={mockData} selectedTrack={1} />
                </Provider>)
          expect(NavigationButtonComponent).toBeTruthy();  
    });

    it("matches snapshot with data", () => {
        const NavigationButtonComponent = render(
            <Provider store={store}>
                <NavigationButtons data={mockData} selectedTrack={1} />)
            </Provider>)
            expect(NavigationButtonComponent).toMatchSnapshot();
    });

    it("matches snapshot without data", () => {
        const NavigationButtonComponent = render(
            <Provider store={store}>
                <NavigationButtons selectedTrack={1} />)
            </Provider>)
            expect(NavigationButtonComponent).toMatchSnapshot();
    });
});
