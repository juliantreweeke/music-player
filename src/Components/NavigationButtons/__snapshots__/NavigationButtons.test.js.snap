// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButtons matches snapshot with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NavigationButtons__Layout-sc-1023fcf-0 eZyLDo"
      >
        <svg
          class="ArrowIcon___StyledSvg-sc-2hfh0e-0 hnmMkd"
          height="32"
          transform="rotate(180)"
          viewBox="0 0 16 16"
          width="32"
        >
          <path
            d="M2 1v14l10-7z"
          />
          <path
            d="M12 1h2v14h-2v-14z"
          />
        </svg>
        <svg
          class="ArrowIcon___StyledSvg-sc-2hfh0e-0 hnmMkd"
          height="32"
          viewBox="0 0 16 16"
          width="32"
        >
          <path
            d="M2 1v14l10-7z"
          />
          <path
            d="M12 1h2v14h-2v-14z"
          />
        </svg>
      </div>
      )
    </div>
  </body>,
  "container": <div>
    <div
      class="NavigationButtons__Layout-sc-1023fcf-0 eZyLDo"
    >
      <svg
        class="ArrowIcon___StyledSvg-sc-2hfh0e-0 hnmMkd"
        height="32"
        transform="rotate(180)"
        viewBox="0 0 16 16"
        width="32"
      >
        <path
          d="M2 1v14l10-7z"
        />
        <path
          d="M12 1h2v14h-2v-14z"
        />
      </svg>
      <svg
        class="ArrowIcon___StyledSvg-sc-2hfh0e-0 hnmMkd"
        height="32"
        viewBox="0 0 16 16"
        width="32"
      >
        <path
          d="M2 1v14l10-7z"
        />
        <path
          d="M12 1h2v14h-2v-14z"
        />
      </svg>
    </div>
    )
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavigationButtons matches snapshot without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NavigationButtons__Layout-sc-1023fcf-0 eZyLDo"
      >
        <svg
          class="ArrowIcon___StyledSvg-sc-2hfh0e-0 eEvyzd"
          height="32"
          transform="rotate(180)"
          viewBox="0 0 16 16"
          width="32"
        >
          <path
            d="M2 1v14l10-7z"
          />
          <path
            d="M12 1h2v14h-2v-14z"
          />
        </svg>
        <svg
          class="ArrowIcon___StyledSvg-sc-2hfh0e-0 eEvyzd"
          height="32"
          viewBox="0 0 16 16"
          width="32"
        >
          <path
            d="M2 1v14l10-7z"
          />
          <path
            d="M12 1h2v14h-2v-14z"
          />
        </svg>
      </div>
      )
    </div>
  </body>,
  "container": <div>
    <div
      class="NavigationButtons__Layout-sc-1023fcf-0 eZyLDo"
    >
      <svg
        class="ArrowIcon___StyledSvg-sc-2hfh0e-0 eEvyzd"
        height="32"
        transform="rotate(180)"
        viewBox="0 0 16 16"
        width="32"
      >
        <path
          d="M2 1v14l10-7z"
        />
        <path
          d="M12 1h2v14h-2v-14z"
        />
      </svg>
      <svg
        class="ArrowIcon___StyledSvg-sc-2hfh0e-0 eEvyzd"
        height="32"
        viewBox="0 0 16 16"
        width="32"
      >
        <path
          d="M2 1v14l10-7z"
        />
        <path
          d="M12 1h2v14h-2v-14z"
        />
      </svg>
    </div>
    )
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
